Formula/pipe-rename.rb

```diff
@@ -0,0 +1,18 @@
+class PipeRename < Formula
+  desc "Rename your files using your favorite text editor"
+  homepage "https://github.com/marcusbuffett/pipe-rename"
+  url "https://crates.io/api/v1/crates/pipe-rename/1.4.2/download"
+  sha256 "9ab27232bc216e23c567ecdcd6f8a590f76a24aa5d9a563e73a21415af82c1df"
+  license "MIT"
+
+  depends_on "rust" => :build
+
+  def install
+    system "tar", "xvf", "pipe-rename-#{version}.crate", "--strip-components=1"
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    system "#{bin}/renamer", "--version"
```

Comment:
We can use Ruby's `PTY` module to emulate a tty.

-----

Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
Is there an existing test that simulates a tty? I don't have much knowledge of ruby, I looked at other formulae as an example when making new ones.

Reply:
OK, I am able to run a no-tty test by manipulating the `EDITOR` env variable. If a single file is passed to `renamer`, this script will output a valid sequence of text to simulate a replace:

```shell
#!/bin/bash

echo "$(cat "$1").txt" > "$1"
```

When combined with `renamer -y`, this will execute a rename automatically.